Microsoft.Azure.Cosmos/src/Resource/ClientContextCore.cs

```diff
@@ -323,6 +324,62 @@ internal override BatchAsyncContainerExecutor GetExecutorForContainer(ContainerC
             return this.batchExecutorCache.GetExecutorForContainer(container, this);
         }
 
+        internal override async Task<Stream> EncryptItemAsync(
+            ContainerCore containerCore,
+            Stream input,
+            EncryptionOptions encryptionOptions,
+            CosmosDiagnosticsContext diagnosticsContext,
+            CancellationToken cancellationToken)
+        {
+            if (encryptionOptions == null)
+            {
+                return input;
+            }
+
+            if (input == null)
+            {
+                throw new ArgumentException(ClientResources.InvalidRequestWithEncryptionOptions);
+            }
+
+            Debug.Assert(containerCore != null);
+            Debug.Assert(diagnosticsContext != null);
+
+            using (diagnosticsContext.CreateScope("Encrypt"))
+            {
+                return await containerCore.ClientContext.EncryptionProcessor.EncryptAsync(
```

Comment:
Please remove the container core from the list of parameters. Just pass the DatabaseCore instead.
```suggestion
                return await this.EncryptionProcessor.EncryptAsync(
```

Reply:
fixed